app/presenters/finder_presenter.rb

```diff
@@ -136,10 +136,7 @@ def sort_options
 
     disabled_option = keywords.blank? ? relevance_sort_option : ''
 
-    selected_option = if values['order']
-                        sort.detect { |option| option['name'].parameterize == values['order'] }
-                      end
-
+    selected_option = values['order'] unless values['order'].nil? && sort.detect { |option| option['name'].parameterize == values['order'] }.nil?
```

Comment:
We could structure the code like so to reduce the line length and improve readability?

```ruby
def sort_values
  sort
    .map { |option| option['name'].parameterize }
    .compact
end

selected_option = values['order'] if sort_values.include?(values['order'])
selected_option ||= default_sort_option_value
```

Reply:
can we add a test for this?

Reply:
@andrewgarner Your suggestion definitely improves on line length and readability, at the expense of performing the computation of `sort_values` regardless of whether of `values['order']` is present. We can definitely improve on the readability though.

@koetsier I'm sure we could

Reply:
Tests added @koetsier

Reply:
I'm not sure this is right:

```selected_option = values['order'] unless values['order'].nil? && sort.detect { |option| option['name'].parameterize == values['order'] }.nil?```

is the same as

```selected_option = values['order'] if !values['order'].nil? || !sort.detect { |option| option['name'].parameterize == values['order'] }.nil?```

so this implies that in the second part ```values['order']``` can be set to nil and so the second part could be simplified to ```sort.any?{| |option| option['name'].nil?}

Reply:
You're right, not the same.

This should actually be `unless ... || ...`. Will update.